add features to the app

hulyak opened the pull request:
## Description

<!-- What does this code change? Why did I choose this approach? Did I learn anything worth sharing? Reminder: This will be a publicly facing representation of your work (READ: help you land that sweet dev gig). -->

- created a custom dropdown with React hooks, created a new file to store the custom hook under `src/hooks/useDetectOutsideClick.js`
- created the dropdown for the Header component, which made the app more user-friendly. If the user wants to share their shopping list with their friends they can toggle it on/off.
- changed the styles of some texts and replaced them with the Typography component from Material UI.
- added media queries for the added functionalities and extra styling. 
- added more information for the user, meaning now the app is displaying what colors mean to the user depending on how soon the user needs to buy the item/items.

Let me know what you think. 




## Related Issue

<!-- If you write "closes" followed by the Github issue number, it will automatically close the issue for you when the PR merges -->

closes #51 

## Acceptance Criteria

<!-- Include AC from the Github issue -->

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<!-- If UI feature, take provide screenshots -->
![Screen Shot 2021-09-08 at 10 05 39 PM](https://user-images.githubusercontent.com/57311842/132626051-4d0faca7-0842-4aec-b466-f0c9467325e9.png)



### After

<!-- If UI feature, take provide screenshots -->
![Screen Shot 2021-09-08 at 10 06 16 PM](https://user-images.githubusercontent.com/57311842/132626086-556616cf-a084-4245-a662-6c9b15a863f0.png)
![Screen Shot 2021-09-08 at 9 55 17 PM](https://user-images.githubusercontent.com/57311842/132625669-3df6f8d5-9ebe-4b63-b16e-12bfa7c8fa75.png)
![Screen Shot 2021-09-08 at 9 55 24 PM](https://user-images.githubusercontent.com/57311842/132625674-4ebee235-81ad-4139-878b-1dbbb350268f.png)
![Screen Shot 2021-09-08 at 9 24 34 PM](https://user-images.githubusercontent.com/57311842/132625681-2f81e523-a1d6-4a49-9f72-f63d9a8b1894.png)
![Screen Shot 2021-09-08 at 9 54 59 PM](https://user-images.githubusercontent.com/57311842/132625697-0f37629d-4b46-4768-bee7-2dbc9dfb200e.png)




## Testing Steps / QA Criteria

<!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->
- pull down 'hulya-custom-dropdown` branch into your local repo
- `npm start`

hulyak committed:
add features to the app

hulyak committed:
add media queries for legend information

hulyak committed:
change where to display the frequency identifiers

isaabutaa commented:
Hi @hulyak! Nice job taking initiative and adding some additional features and styles to the app. I only have a couple critiques, but nothing major. 

First, I really like the "Would you like to invite your friends to your shopping list?" idea. I know TCL-27 did that, and it's a nice touch. If the user hovers over that text, the cursor changes to a pointer, so they know they can click on it, but I wonder if maybe we can add a small border, or slightly different background color with a drop shadow to make it more obvious that it's clickable? 

Second, on the Items List page, I noticed that you added the list items that act as a reference so the user knows which color means what — I think that's a good idea. At first, I thought that it was the Add Item form because it looks very similar, but then I realized it was different. I don't know if I have any actionable recommendations for that, but it's just a thought I had.

Also, I realized that the font we're importing, Asap, isn't the font that's on the app. I think it's just sans-serif. If that's something we want to change, my recommendation would be to copy + paste the Google fonts link in the header in public/index.html, and then put the font in the styles page, or import the font in the styles page:

Link in header in public/index.html:
```
<header>
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Asap&display=swap" rel="stylesheet">
</header>
```
Import in root css file:
```
@import url('https://fonts.googleapis.com/css2?family=Asap&display=swap');
```
And then to actually use the font, the simplest solution might be to do `font-family: 'Asap', sans-serif;` in the body selector of the root css file.

Those are just some of my thoughts, but again, way to take initiative. I know we ran out of time last week, and I think it's totally valid to want to add more to the app. 👍

hulyak committed:
add border for token prompt and change styles

hulyak commented:
Thanks, @isaabutaa for reviewing the PR and writing a very good review. I checked the font, it's actually ASAP. 
![Screen Shot 2021-09-09 at 3 17 08 PM](https://user-images.githubusercontent.com/57311842/132787074-143fb92c-a95e-464b-9cef-3bb085d1c3b4.png)

I have added a border for the token prompt and changed some styles. You can check out the new version. Thanks!

luisaugusto commented on src/App.css:
```diff
@@ -2,21 +2,31 @@
+  display: inline-block;
+  transform: translateX(10%);
 }
 
-.App-link {
-  color: #09d3ac;
+@media only screen and (max-width: 700px) {
+  .menu.active {
+    transform: translateX(1%);
+  }
```
I checked out the app since I was a little unsure of that the translates were doing here, and I think it's more effective to instead use flexbox containers that can align content without the need for media queries. You can remove any translate properties add something like this to `.menu-container`:

```
display: flex;
flex-direction: column;
align-items: center;
```

hulyak replied:
thank you Luis for reviewing the PR, I didn't want to disturb the mentors. Thanks, again! I have deleted the transforms and added flexbox.

luisaugusto commented on src/hooks/useDetectOutsideClick.js:
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react';
```
Very nice job, making a custom hook for this!

luisaugusto commented on src/themes/theme.js:
```diff
@@ -42,9 +42,30 @@ export const theme = createTheme({
         fontSize: '4rem',
       },
     },
+    h3: {
```
I like how the styles for the headers are in the theme, and you are using the `Typography` component to render them. This is a much nicer approach than using global styles for the header tags 🎉

luisaugusto reviewed (approved):
The app is looking awesome, nice work with these changes!

hulyak committed:
update styles
